File: postgres/sample_pg.pl
#!/usr/bin/perl
use strict;
use warnings;
 
#-- Perl script that uses the "psql" client utility to
#-- execute an SQL command

#-- Name of the database involved
my $database = $ARGV[0];
if (not defined $database)
{
  print "**Error: missing database name argument\n";
  print "  I quit!\n";
  exit;
}

#-- SQL command
my $qry = 'select city,population from canada_city where population>100000 order by population desc';

#-- Open the "psql" client utility with the -B option to return
#-- results in batch tab-delimited format

#my $command = "mysql -B -D ${database} -e '${qry}'";
my $command = "psql -t -AF \$'\t' -c '${qry}'";

my $line;
my @row;
my @fields;

print "Query result\n";
print '-'x50,"\n";

open(COMM,"${command}|");
while ($line=<COMM>)
{
  chop $line;
  {
    @row = split(/\t/,$line);
    print "Row: @row","\n";
  }
}
close(COMM);

exit;

File: postgres/sample_pg.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
 
import psycopg2
import sys
 
 
con = None

import os
print(os.environ['USER'])

user = os.environ['USER']
database = "%s_db" % user

print("user:" + user)
print("database:" + database)
 
try:
    con = psycopg2.connect("host='cedar-pgsql-vm.int.cedar.computecanada.ca' dbname='" + database + "' user='" + user + "'")   
    cur = con.cursor()
    cur.execute("select city,population from canada_city where population>100000 order by population desc")
 
    while True:
        row = cur.fetchone()
 
        if row == None:
            break
 
        print("City: " + row[0] + "\t\tPopulation: " + str(row[1]))
 
except psycopg2.DatabaseError, e:
    if con:
        con.rollback()
 
    print 'Error %s' % e    
    sys.exit(1)
 
finally:   
    if con:
        con.close()
